src/alm.jl

```diff
@@ -108,6 +112,90 @@ function readAlmFromFITS(fileName, t::Type{T}) where {T <: Complex}
 end
 end
 
+###############################################################
+
+"""
+    almExplicitIndex(alms::Alm{Complex{T}}; lmax::Integer = -1, nmax::Integer = -1) where {T <: Number} -> Vector{Int}
+    almExplicitIndex(lmax::Integer, mmax::Integer) -> Vector{Int}
+
+Compute the explicit index scheme, i.e. ``index = \\ell^2 + \\ell + m + 1`` up to a
+certain ``ℓ`` and ``m`` if specified (and non-negative).
+
+"""
+
+function almExplicitIndex(alm::Alm{Complex{T}}; lmax::Integer = -1, mmax::Integer = -1) where {T <: Number}
+    if lmax < 0
+        lmax = alm.lmax
+    end
+    if mmax < 0
+        mmax = alm.mmax
+    end
+    idx = Vector{Int}()
+    for m = 0:mmax
+        for l = m:lmax
+            i = l^2 + l + m + 1
+            append!(idx, i)
+        end
+    end
+    return idx
+end
```

Comment:
This code is quite inefficient, as the array needs to grow progressively every time `append!` is invoked. Also, it's better to avoid the parameter `alm` to be passed, as the only meaningful inputs for this function are `lmax` and `mmax`.

The following code should work better:

```julia
# No need to specify the type for `lmax` and `mmax`
function almExplicitIndex2(lmax, mmax) where {T <: Number}
    # Step 1: count the number of elements in the output
    count = 0
    for m = 0:mmax
        (lmax ≥ m) && (count += (lmax - m) + 1)
    end

    # Step 2: allocate the vector for the output in one batch
    idx = Vector{Int}(undef, count)

    # Step 3: fill the output
    i = 1
    for m = 0:mmax
        for l = m:lmax
            idx[i] = l^2 + l + m + 1
            i += 1
        end
    end

    idx  # In Julia it's preferable to avoid "return" at the end of functions
end
```

I did a few measurements and I see a ~10 time increase in speed and a significant reduction in allocated memory.

Original implementation:

```
julia> using BenchmarkTools
julia> @benchmark almExplicitIndex(1000, 3000)
BenchmarkTools.Trial: 935 samples with 1 evaluation.
 Range (min … max):  4.547 ms …   6.836 ms  ┊ GC (min … max): 0.00% … 21.34%
 Time  (median):     5.435 ms               ┊ GC (median):    0.00%
 Time  (mean ± σ):   5.308 ms ± 631.319 μs  ┊ GC (mean ± σ):  1.56% ±  5.03%

   ▇█▄                        ▅▃▁                              
  ▇████▄▃▂▂▂▁▂▂▂▂▄▄▄▃▂▂▂▃▅▃▃▅████▆▅▃▃▃▂▂▂▂▂▂▁▂▂▁▂▁▁▄▄▃▃▃▃▄▃▄▃ ▃
  4.55 ms         Histogram: frequency by time        6.69 ms <

 Memory estimate: 5.23 MiB, allocs estimate: 13.
```

New implementation:

```
julia> @benchmark almExplicitIndex2(1000, 3000)
BenchmarkTools.Trial: 8343 samples with 1 evaluation.
 Range (min … max):  509.585 μs …   1.947 ms  ┊ GC (min … max): 0.00% … 29.00%
 Time  (median):     544.338 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):   593.617 μs ± 154.445 μs  ┊ GC (mean ± σ):  7.20% ± 13.30%

  ▃█▆▇▆▆▅▄▃▂▁                                        ▁▁▂▂▂▂▁▁▁  ▂
  ████████████▇▇▅▄▇▆▅▅▃▃▅▄▄▅▃▁▃▁▄▄▁▃▄▅▄▅▅▆▄▅▅▅▄▃▄▄▄▅▅██████████ █
  510 μs        Histogram: log(frequency) by time       1.11 ms <

 Memory estimate: 3.83 MiB, allocs estimate: 2.

Reply:
Thanks for the hint! I've fixed it with a new commit.